libpromises/evalfunction.c

```diff
@@ -6659,6 +6658,72 @@ static FnCallResult FnCallReadData(ARG_UNUSED EvalContext *ctx,
     return (FnCallResult) { FNCALL_SUCCESS, (Rval) { json, RVAL_TYPE_CONTAINER } };
 }
 
+static FnCallResult FnCallClassExprFilter(EvalContext *ctx,
+                                          ARG_UNUSED Policy const *policy,
+                                          ARG_UNUSED FnCall const *fp,
+                                          Rlist const *args)
+{
+    char const *path = RlistScalarValue(args);
+    size_t const col = IntFromString(RlistScalarValue(args->next));
+    char const *delim = RlistScalarValue(args->next->next);
+    bool has_heading = BooleanFromString(RlistScalarValue(args->next->next->next));
+    int const sort_col = IntFromString(RlistScalarValue(args->next->next->next->next));
+
```

Comment:
Would something like having an `Rlist const *arg = args` and then doing `RlistScalarValue(arg = arg->next)` several times instead be easier to read and manage than increasing the number of `->next` calls in each line?